test/test_datasets/test_conceptual_captions.py

```diff
@@ -0,0 +1,85 @@
+import io
+import urllib
+from typing import Any, Iterable
+from unittest.mock import patch
+
+import numpy as np
+import requests
+from PIL import Image
+from squirrel_datasets_core.datasets.conceptual_captions.driver import CC12MDriver
+
+from mock_utils import create_random_str
+
+
+class MockImageResponse:
+    calls = 0
+
+    def read(self) -> bytes:
+        """Mock reading file"""
+        MockImageResponse.calls += 1
+        if MockImageResponse.calls == 1:
+            raise urllib.error.HTTPError(None, None, None, None, None)
+
+        rand_img = np.random.randint(0, 255, size=(10, 10, 3)).astype(np.uint8)
+        img_byte_arr = io.BytesIO()
+        Image.fromarray(rand_img).save(img_byte_arr, format="PNG")
+        img_byte_arr.seek(0)
+        return img_byte_arr.read()
+
+
+class MockTextResponse:
+    N_SAMPLES = 10
+
+    def iter_lines(self) -> Iterable:
+        """Mock iteratively fetching lines from web document"""
+        for _ in range(MockTextResponse.N_SAMPLES):
+            yield f"https://{create_random_str()}\t{create_random_str()}".encode("utf-8")
```

Comment:
Can make what each str chunk holds more explicit
```suggestion
            url, caption = f"https://{create_random_str()}", create_random_str()
            yield f"{url}\t{caption}".encode("utf-8")
```